CMakeLists.txt

```diff
@@ -0,0 +1,470 @@
+  add_test(NAME issue178_test COMMAND issue178_test)
+
+  add_executable(issue200_test
+    "${PROJECT_SOURCE_DIR}/issues/issue200_test.cc"
+  )
+  target_link_libraries(issue200_test leveldb_test_base)
+  add_test(NAME issue200_test COMMAND issue200_test)
+
+  add_executable(filter_block_test
+    "${PROJECT_SOURCE_DIR}/table/filter_block_test.cc"
+  )
+  target_link_libraries(filter_block_test leveldb_test_base)
+  add_test(NAME filter_block_test COMMAND filter_block_test)
+
+  add_executable(table_test
+    "${PROJECT_SOURCE_DIR}/table/table_test.cc"
+  )
+  target_link_libraries(table_test leveldb_test_base)
+  add_test(NAME table_test COMMAND table_test)
+
+  add_executable(arena_test
+    "${PROJECT_SOURCE_DIR}/util/arena_test.cc"
+  )
+  target_link_libraries(arena_test leveldb_test_base)
+  add_test(NAME arena_test COMMAND arena_test)
+
+  add_executable(bloom_test
+    "${PROJECT_SOURCE_DIR}/util/bloom_test.cc"
+  )
+  target_link_libraries(bloom_test leveldb_test_base)
+  add_test(NAME bloom_test COMMAND bloom_test)
+
+  add_executable(cache_test
+    "${PROJECT_SOURCE_DIR}/util/cache_test.cc"
+  )
+  target_link_libraries(cache_test leveldb_test_base)
+  add_test(NAME cache_test COMMAND cache_test)
+
+  add_executable(coding_test
+    "${PROJECT_SOURCE_DIR}/util/coding_test.cc"
+  )
+  target_link_libraries(coding_test leveldb_test_base)
+  add_test(NAME coding_test COMMAND coding_test)
+
+  add_executable(crc32c_test
+    "${PROJECT_SOURCE_DIR}/util/crc32c_test.cc"
+  )
+  target_link_libraries(crc32c_test leveldb_test_base)
+  add_test(NAME crc32c_test COMMAND crc32c_test)
+
+  add_executable(env_posix_test
+    "${PROJECT_SOURCE_DIR}/util/env_posix_test_helper.h"
+    "${PROJECT_SOURCE_DIR}/util/env_posix_test.cc"
+  )
+  target_link_libraries(env_posix_test leveldb_test_base)
+  add_test(NAME env_posix_test COMMAND env_posix_test)
+
+  add_executable(env_test
+    "${PROJECT_SOURCE_DIR}/util/env_test.cc"
+  )
+  target_link_libraries(env_test leveldb_test_base)
+  add_test(NAME env_test COMMAND env_test)
+
+  add_executable(hash_test
+    "${PROJECT_SOURCE_DIR}/util/hash_test.cc"
+  )
+  target_link_libraries(hash_test leveldb_test_base)
+  add_test(NAME hash_test COMMAND hash_test)
+endif(LEVELDB_BUILD_TESTS)
+
+if(LEVELDB_BUILD_BENCHMARKS)
+  add_library(leveldb_bench_base "")
+  target_sources(leveldb_bench_base
+    PRIVATE
+      "${PROJECT_SOURCE_DIR}/util/histogram.cc"
+      "${PROJECT_SOURCE_DIR}/util/testutil.cc"
+    PUBLIC
+      "${PROJECT_BINARY_DIR}/include/port/port_config.h"
+      "${PROJECT_SOURCE_DIR}/util/histogram.h"
+      "${PROJECT_SOURCE_DIR}/util/testutil.h"
+  )
+  target_compile_definitions(leveldb_bench_base
+    PUBLIC
+      LEVELDB_PLATFORM_POSIX
+  )
+  target_link_libraries(leveldb_bench_base leveldb)
+
+  add_executable(db_bench
+    "${PROJECT_SOURCE_DIR}/db/db_bench.cc"
+  )
+  target_link_libraries(db_bench leveldb_bench_base)
+
+  check_library_exists(sqlite3 sqlite3_open "" HAVE_SQLITE3)
```

Comment:
If this code planned to be used while building with Hunter then should be managed by `hunter_add_package`.

Reply:
Right. Seeing as I think this is only for the benchmarking, I am not sure if better to fix or remove.

Reply:
You can keep it, just check that `LEVELDB_BUILD_BENCHMARKS` is `OFF` by default or add `LEVELDB_BUILD_BENCHMARKS=OFF` explicitly:
* https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#cmake-options